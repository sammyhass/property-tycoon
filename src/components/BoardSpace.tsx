import { AspectRatio, Text } from '@chakra-ui/react';
import React from 'react';

const BASE_SIZE = 120;
interface BoardSpaceProps {
  // big only on corners
  big?: boolean;
}

export default function BoardSpace({ big = false }: BoardSpaceProps) {
  return (
    <AspectRatio ratio={big ? 1 : 2} h={`${BASE_SIZE}px`} w={'auto'}>
      <Text>Board</Text>
    </AspectRatio>
  );
}
