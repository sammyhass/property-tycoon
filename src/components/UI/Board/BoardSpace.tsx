import { Box } from '@chakra-ui/react';
import { BoardSpace, CardType, SpaceType } from '@prisma/client';
import React from 'react';

function rad(deg: number) {
  return (deg / 180) * Math.PI;
}

const EXAMPLE_SPACES: BoardSpace[] = [
  {
    // SpaceType = SpaceType.TAKE_CARD - take_card field is non-null and contains card type to pick up
    board_position: 0,
    game_id: '',
    created_at: new Date(),
    take_card: CardType.POT_LUCK,
    property_id: '',
    space_type: SpaceType.TAKE_CARD,
    locked: false,
  },
  {
    // SpaceType = FreeParking,
    board_position: 1,
    game_id: '',
    take_card: null,
    created_at: new Date(),
    property_id: null,
    space_type: SpaceType.FREE_PARKING,
    locked: false,
  },
  {
    // SpaceType = Property, property_id will be the id of the property present
    board_position: 2,
    game_id: '',
    take_card: null,
    locked: false,
    created_at: new Date(),
    property_id: '1',
    space_type: SpaceType.PROPERTY,
  },
  {
    // SpaceType = EMPTY, just show empty box
    space_type: 'EMPTY',
    board_position: 3,
    game_id: '',
    take_card: null,
    created_at: new Date(),
    locked: false,
    property_id: null,
  },
];

// Todo: board cell  should take a board_space as props, which will include ALL necessary information to render the correct tile
export default function BoardSpaceView({
  board_position,
  created_at,
  game_id,
  property_id,
  space_type,
  take_card,
}: BoardSpace) {
  return <Box></Box>;
}
// let boardRadius = (boardSize - 1) / 2;
// let totalCells = boardSize * 4 - 4;
// let boardSqr = boardRadius * Math.sqrt(2);

// function calculateRow(n: number) {
//   return Math.round(
//     Math.min(
//       Math.max(
//         boardSqr * Math.sin(rad((n / totalCells) * 360 - 135)) + boardRadius,
//         0
//       ),
//       boardSize - 1
//     )
//   );
// }

// function calculateColumn(n: number) {
//   return Math.round(
//     Math.min(
//       Math.max(
//         boardSqr * Math.cos(rad((n / totalCells) * 360 - 135)) + boardRadius,
//         0
//       ),
//       boardSize - 1
//     )
//   );
// }

// return (
//   <Box
//     gridRow={calculateRow(n) + 1}
//     gridColumn={calculateColumn(n) + 1}
//     minH={'70px'}
//     w={`${cellSize}px`}
//     _hover={{
//       transform: `scale(1)`,
//     }}
//     bg="blue"
//     onClick={() => onTileClick && onTileClick(n)}
//   >
//     <Text>{n + 1}</Text>
//   </Box>
// );
