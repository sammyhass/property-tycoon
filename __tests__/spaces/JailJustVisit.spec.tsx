import BoardSpace from '@/components/Game/Board/spaces';
import { render } from '@testing-library/react';
import React from 'react';

describe('Just Visiting/Jail Board Space', () => {
  it('matches snapshot', () => {
    const { container } = render(<BoardSpace.Jail />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the correct text', () => {
    const { container } = render(<BoardSpace.Jail />);
    expect(container.textContent).toContain('Just');
    expect(container.textContent).toContain('Visiting');
  });
});
