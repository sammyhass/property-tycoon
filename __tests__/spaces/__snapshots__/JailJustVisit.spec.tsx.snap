// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Just Visiting/Jail Board Space matches snapshot 1`] = `
<div
  class="boardSpace square inJail"
>
  <div
    class="just"
  >
    Just
  </div>
  <div
    class="visiting"
  >
    Visiting
  </div>
  <div
    class="inner"
  >
    <div>
      In
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-n2bes8"
        >
          ⛓️
        </p>
      </div>
      Jail
    </div>
  </div>
</div>
`;
